HipercowApi/Controllers/ClusterAccessController.cs

```diff
@@ -2,62 +2,49 @@
 
 namespace HipercowApi.Controllers
 {
-    using HipercowApi.Models;
+    using Hipercow_api.Tools.Exceptions;
     using HipercowApi.Tools;
-    using Microsoft.AspNetCore.Authorization;
     using Microsoft.AspNetCore.Mvc;
 
     /// <summary>
     /// The /clusteraccess endpoint requires an authentication
     /// token, and provides the list of clusters that a certain
     /// user has access to.
     /// </summary>
+    /// <remarks>
+    /// Initializes a new instance of the <see cref="ClusterAccessController"/> class.
+    /// </remarks>
+    /// <param name="jwtSupport">Our class for encrypting JWTs.</param>
     [ApiController]
     [Route("api/v1/[controller]")]
-    public class ClusterAccessController : ControllerBase
+    public class ClusterAccessController(JwtSupport jwtSupport) : ControllerBase
     {
-        private readonly UserSessionManager sessionManager;
-
-        /// <summary>
-        /// Initializes a new instance of the
-        /// <see cref="ClusterAccessController"/> class.
-        /// </summary>
-        /// <param name="sessionManager">The session manager so we can look up the
-        /// login details of previous sessions in the memory cache.
-        /// </param>
-        public ClusterAccessController(
-            UserSessionManager sessionManager)
-        {
-            this.sessionManager = sessionManager;
-        }
+        private readonly JwtSupport _jwtSupport = jwtSupport;
 
         /// <summary>
         /// Authenticated endpoint to return the list of clusters available to a user.
         /// </summary>
-        /// <param name="sessionId">Session Id for retrieving login details.</param>
+        /// <param name="authHeader">Header information to get the encrypted JWT from.</param>
         /// <returns>
         /// A list of cluster names.
         /// </returns>
-        [Authorize]
         [HttpGet]
-        public IActionResult GetMyClusters([FromHeader(Name = "X-Session-Id")] string sessionId)
+        public IActionResult GetMyClusters([FromHeader(Name = "Authorization")]
+                                           string authHeader)
         {
-            string? jwtUsername = this.User.Identity!.Name;
-            UserSession? session = this.sessionManager.RetrieveSession(sessionId);
-            IActionResult? result = Utils.CheckTokenAndSession(this, jwtUsername, session);
-            if (result is not null)
-            {
-                return result;
-            }
+            var token = authHeader.Replace("Bearer ", string.Empty);
+            var dict = _jwtSupport.DecryptToken(token);
```

Comment:
I feel like if there's any other controllers which need to decrypt the tokens then we should probably get that happening in middleware too rather than forcing every controller wihch needs user information to do it.

Reply:
Middleware for JWT done...